bot/roles.go

```diff
@@ -0,0 +1,55 @@
+package bot
+
+import (
+	"fmt"
+	"log"
+	"maps"
+	"slices"
+	"strings"
+)
+
+func roleListCmd(ctx CommandCtx) error {
+	rolesString := strings.Join(slices.Collect(maps.Keys(joinableRolesMap)), ", ")
+	if err := ctx.Reply(fmt.Sprintf("Available roles: %s.\n Available commands: !rjoin & !rleave.", rolesString)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func roleJoinCmd(ctx CommandCtx) error {
+	args := ctx.Args
+	msg := ctx.Message
+	sess := ctx.Session
+	if roleID, exists := joinableRolesMap[args[0]]; exists {
+		err := sess.GuildMemberRoleAdd(msg.GuildID, msg.Author.ID, roleID)
+		if err != nil {
+			log.Printf("error adding role: %s", err)
+			return fmt.Errorf("error adding user to role: %w", err)
+		} else {
+			log.Printf("Added user with id: %s (%s) to %s role", msg.Author.ID, msg.Author.Username, roleID)
+			if err := ctx.Reply(fmt.Sprintf("You've been added to the group %s.", args[0])); err != nil {
+				return err
+			}
+			return nil
+		}
+	}
+	return nil
+}
+
+func roleLeaveCmd(ctx CommandCtx) error {
+	args := ctx.Args
+	msg := ctx.Message
+	if roleID, exists := joinableRolesMap[args[0]]; exists {
+		err := ctx.Session.GuildMemberRoleRemove(msg.GuildID, msg.Author.ID, roleID)
+		if err != nil {
+			return fmt.Errorf("error removing role: %w", err)
+		} else {
```

Comment:
dont need the else here, keeps the following code outside of a block/one indent level less.